benchmark/benchmark.py

```diff
@@ -0,0 +1,117 @@
+"""Benchmarking utility for PennyLane.
+"""
+# pylint: disable=import-outside-toplevel,invalid-name
+import functools
+import importlib
+import subprocess
+import sys
+
+import numpy as np
+
+import pennylane as qml
+
+
+def timing(func, *, number=10, repeat=5):
+    """Time the given function.
+
+    Args:
+        func (callable[None, Any]): function to time
+        number (int): number of loops per test
+        repeat (int): the number the timing test is run
+    """
+    import timeit
```

Comment:
I assume you went with in-function imports in order to minimize overhead for running specific benchmarks? i.e., to get more accurate estimates of actual pennylane runtimes?

Reply:
Not really, the timing should only start after the imports are done. It's more about explicitly pointing out that the three functions use three different benchmarking modules (timeit, cProfile, perfplot), and only one of them is executed during each run.

Reply:
Got it, makes sense